airbyte-ci/connectors/auto_merge/src/auto_merge/main.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from __future__ import annotations
+
+import logging
+import os
+import time
+from collections.abc import Iterable, Iterator
+from contextlib import contextmanager
+from typing import TYPE_CHECKING
+
+from github import Auth, Github
+
+from .consts import AIRBYTE_REPO, AUTO_MERGE_LABEL, BASE_BRANCH, CONNECTOR_PATH_PREFIXES
+from .env import GITHUB_TOKEN, PRODUCTION
+
+if TYPE_CHECKING:
+    from github.Commit import Commit as GithubCommit
+    from github.File import File as GithubFile
+    from github.PullRequest import PullRequest
+    from github.Repository import Repository as GithubRepo
+
+logging.basicConfig()
+logging.getLogger().setLevel(logging.INFO)
+
+
+@contextmanager
+def github_client() -> Iterator[Github]:
+    client = None
+    try:
+        client = Github(auth=Auth.Token(GITHUB_TOKEN), seconds_between_requests=0)
+        yield client
+    finally:
+        if client:
+            client.close()
+
+
+def check_if_modifies_connectors_only(modified_files: Iterable[GithubFile]) -> bool:
+    """Check if all modified files are in CONNECTOR_PATH_PREFIXES
+
+    Args:
+        modified_files (Iterable[GithubFile]): All the modified files on a PR.
+
+    Returns:
+        bool: True if all modified files are in CONNECTOR_PATH_PREFIXES, False otherwise
+    """
+    for file in modified_files:
+        if not any(file.filename.startswith(prefix) for prefix in CONNECTOR_PATH_PREFIXES):
```

Comment:
❓ Didn't we have this same logic in the connector ops package? If so should we introduce that as dependency so if we change where connectors are located we dont have to update so many places?